src/views/Folders.vue

```diff
@@ -29,14 +29,24 @@
 	<EmptyContent v-else-if="error">
 		{{ t('photos', 'An error occurred') }}
 	</EmptyContent>
+	<NcEmptyContent v-else-if="initializing" icon="icon-loading">
+		{{ t('photos', 'Loading folders') }}
+	</NcEmptyContent>
 
 	<!-- Folder content -->
-	<div v-else-if="!loading">
-		<Navigation v-if="folder"
-			key="navigation"
-			v-bind="folder"
+	<div v-else-if="!initializing">
+		<HeaderNavigation key="navigation"
+			:loading="loading"
+			:path="folder.filename"
+			:title="folder.basename"
 			:root-title="rootTitle"
-			:show-actions="true" />
+			@refresh="onRefresh">
+			<UploadPicker :accept="allowedMimes"
+				:destination="folder.filename"
+				:multiple="true"
+				@uploaded="onRefresh" />
```

Comment:
Every uploaded file request a refresh. For big folders, that could be expensive, maybe we could fetch the newly uploaded file only? Not sure the albums API allows it :/

Reply:
This is to progressively display the files will they are uploaded ?

Reply:
Shouldn't it be enough to add the file id to the folder list without refreshing the full view?

Reply:
We don't know the File id without fetching the content :see_no_evil: 

> This is to progressively display the files will they are uploaded ?

This or update the view after upload

Reply:
Might be best to do a propfind on the new file, this way we are sure to have all the data, and less expensive than propfinding all the list. This is what files is doing apparently.